[RS-2387] Application Layer Policies to support HTTP headers - changes to docs. (#2071)
tigera/docs: calico-enterprise/network-policy/application-layer-policies/alp.mdx

You can restrict ingress traffic using HTTP match criteria using Global network policy. 
For a list of all HTTP match parameters, see [Global network policy](/reference/resources/globalnetworkpolicy.mdx).

> In the following example, the trading app is allowed ingress traffic only for HTTP GET requests that match the exact path `/projects/calico`, or that begins with the prefix `/users`.

Reviewer: ```suggestion
In the following example, the trading app is allowed ingress traffic only for HTTP GET requests that match the exact path `/projects/calico`, or that begin with the prefix `/users`.
```

> Additionally the HTTP request must contain the `Content-Type` HTTP header with a value of either `text/plain` or `application/json`, as well as the `User-Agent` HTTP header with a value having prefix `curl`.

Reviewer: ```suggestion
Additionally, the HTTP request must contain the `Content-Type` HTTP header with a value of either `text/plain` or `application/json`, as well as the `User-Agent` HTTP header with a value that has the prefix `curl`.
```

> Any HTTP request from the trading app that does not fulfill the outlined criteria would be rejected.

Reviewer: ```suggestion
Any HTTP request from the trading app that does not fulfill the outlined criteria is rejected.
```
